lib/afmotion/http_client.rb

```diff
@@ -62,6 +65,12 @@ def request_serializer(serializer)
       else
         @operation_manager.requestSerializer = serializer
       end
+      reinit_options
+    end
+
+    def reinit_options
+      @headers.each{ |h,v| header(h, v) } if !@headers.nil? && !@headers.empty?
```

Comment:
Also, the check for `empty?` is redundant (`each` will call the block 0 times if it's empty)

So I guess it would just be `unless @headers.nil?` or `if @headers`